src/SurfaceFluxes.jl

```diff
@@ -54,417 +57,752 @@ Surface flux conditions, returned from `surface_conditions`.
+        )
+    if Δθ ≈ FT(0) || windspeed(sc) ≈ FT(0)
+        Ch = FT(0)
+    else
+        Ch = ustar * θstar / windspeed(sc) / Δθ
+    end
+    return Ch
+end
 
-Returns u_star, θ_star, ... given equations (17), (18) for FV and (8), (9) for DG
-from Nishizawa & Kitamura (2018).
+"""
+  heat_exchange_coefficient(param_set, L_MO, sc, scheme)
+Return Ch, the heat exchange coefficient given the
+Monin-Obukhov lengthscale.
+"""
+function heat_exchange_coefficient(param_set, L_MO, sc::Coefficients, scheme)
+    return sc.Ch
+end
 
+"""
+  momentum_fluxes(param_set, Cd, sc, scheme)
+Compute and return the momentum fluxes
 ## Arguments
- - `uf` universal function family
- - `z_in` Input height for the similarity functions. It is Δz for FV and the height
-    of the first nodal point for DG
- - `z_0` roughness lengths for state variable array `x`
- - `x_in` Inner values for state variable array `x`. They correspond to volume averages in FV and
-    nodal point values in DG
- - `x_s` surface values for state variable array `x`
- - `transport` Type of transport (momentum or heat)
-
-"""
-function compute_physical_scale(
-    uf::UF.AbstractUniversalFunction{FT},
-    z_in,
-    z_0,
-    x_in,
-    x_s,
-    transport,
-    ::FVScheme,
-) where {FT}
-    von_karman_const::FT = CPSGS.von_karman_const(uf.param_set)
-    _π_group = FT(UF.π_group(uf, transport))
-    _π_group⁻¹ = (1 / _π_group)
-    R_z0 = 1 - z_0 / z_in
-    temp1 = log(z_in / z_0)
-    temp2 = -UF.Psi(uf, z_in / uf.L, transport)
-    temp3 = z_0 / z_in * UF.Psi(uf, z_0 / uf.L, transport)
-    temp4 = R_z0 * (UF.psi(uf, z_0 / uf.L, transport) - 1)
-    Σterms = temp1 + temp2 + temp3 + temp4
-    return _π_group⁻¹ * von_karman_const / Σterms * (x_in - x_s)
+  - param_set: Abstract Parameter Set containing physical, thermodynamic parameters.
+  - Cd: Momentum exchange coefficient
+  - sc: Container for surface conditions based on known combination
+        of the state vector, and {fluxes, friction velocity, exchange coefficients} for a given experiment
+  - scheme: Discretisation scheme (currently supports DG and FV)
+"""
+function momentum_fluxes(param_set, Cd, sc::AbstractSurfaceConditions, scheme)
+    ρ_sfc = TD.air_density(ts_sfc(sc))
+    ρτxz = -ρ_sfc * Cd * Δu1(sc) * windspeed(sc)
+    ρτyz = -ρ_sfc * Cd * Δu2(sc) * windspeed(sc)
+    return (ρτxz, ρτyz)
 end
 
-function compute_physical_scale(
-    uf::UF.AbstractUniversalFunction{FT},
-    z_in,
-    z_0,
-    x_in,
-    x_s,
-    transport,
-    ::DGScheme,
-) where {FT}
-    von_karman_const::FT = CPSGS.von_karman_const(uf.param_set)
-    _π_group = FT(UF.π_group(uf, transport))
-    _π_group⁻¹ = (1 / _π_group)
-    temp1 = log(z_in / z_0)
-    temp2 = -UF.psi(uf, z_in / uf.L, transport)
-    temp3 = UF.psi(uf, z_0 / uf.L, transport)
-    Σterms = temp1 + temp2 + temp3
-    return _π_group⁻¹ * von_karman_const / Σterms * (x_in - x_s)
+"""
+  sensible_heat_flux(param_set, Ch, sc, scheme)
+In cases where surface fluxes are known,
```

Comment:
Not sure what you mean here. The surface fluxes do not seem to be known, but surface conditions (e.g., SST) are.